src/imports/background/connection-handler.js

```diff
@@ -1,6 +1,7 @@
 import { CMDS, DEF_CONCURRENCY } from 'src/options/imports/constants'
 import ProgressManager from './progress-manager'
 import stateManager from './state-manager'
+import { parseFile } from 'src/imports/background/utils'
```

Comment:
I would prefer this file parsing stuff not be in here, but instead in the `imports/background/data-sources` module to be consistent with the other external data stuff. If you look at the [imports arch overview](https://github.com/WorldBrain/Memex/blob/develop/src/imports/imports-arch.png) that `Web Ext APIs` is like an external side-effect that affects what's how the entire imports system will behave, and `DataSources` acts like an interface to it. Doing this essentially adds another yellow box pointing to `ConnectionHandler`. Though I think you would still need to pass down the URL, but you can get rid of all the `data` args.

I'd actually now like to redesign it a bit to eliminate those side-effects completely and just have everything passed down when the system is set up, but that's a bit more work.